plugins/modules/identity/sap_user.py

```diff
@@ -0,0 +1,510 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2021, Rainer Leber <rainerleber@gmail.com> <rainer.leber@sva.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: sap_user
+short_description: This module will manage a user entities in a SAP S4/HANA environment.
+version_added: "1.1.0"
+description:
+  - The C(sap_user) module depends on C(pyrfc) Python library (version 2.4.0 and upwards).
+    Depending on distribution you are using, you may need to install additional packages to
+    have these available.
+  - This module will use the following user BAPIs to manage user entities.
+    - C(BAPI_USER_GET_DETAIL)
+    - C(BAPI_USER_DELETE)
+    - C(BAPI_USER_CREATE1)
+    - C(BAPI_USER_CHANGE)
+    - C(BAPI_USER_ACTGROUPS_ASSIGN)
+    - C(BAPI_USER_PROFILES_ASSIGN)
+    - C(BAPI_USER_UNLOCK)
+    - C(BAPI_USER_LOCK)
+options:
+    state:
+        description:
+          - The decission what to do with the user.
+          - Could be C('present'), C('absent'), C('lock'), C('unlock')
+        default: 'present'
+        choices:
+          - 'present'
+          - 'absent'
+          - 'lock'
+          - 'unlock'
+        required: false
+        type: str
+    force:
+        description:
+          - Must be c('True') if the password or type should be overwritten.
+        default: False
+        required: False
+        type: bool
+    conn_username:
+        description: The required username for the SAP system.
+        required: true
+        type: str
+    conn_password:
+        description: The required password for the SAP system.
+        required: true
+        type: str
+    host:
+        description: The required host for the SAP system. Can be either an FQDN or IP Address.
+        required: true
+        type: str
+    sysnr:
+        description:
+          - The system number of the SAP system.
+          - You must quote the value to ensure retaining the leading zeros.
+        default: '00'
+        type: str
+    client:
+        description:
+          - The client number to connect to.
+          - You must quote the value to ensure retaining the leading zeros.
+        default: '000'
+        type: str
```

Comment:
required?

Reply:
because of the default there is no need to specifiy it